File: python/lpthw/fib.c
#include <stdio.h>
#include <stdlib.h>

int fib(int n)
{
	if( n==0 || n ==1)
		return 1;
	else
		return fib(n-1) +fib(n-2);
}


int main(int argc, char **argv)
{
	if(argc != 2)
	{
		fprintf(stderr, "usage: %s range\n", argv[0]);
		exit(0);
	}

	printf("Result is: %d\n", fib(atoi(argv[1])) );
	return 0;

}

File: python/lpthw/fib.java
public class fib
{

	private static int calFib(int number)
	{
		if( number ==0 || number ==1)
			return 1;
		else
			return calFib(number-1) + calFib(number-2);
	}

	public static void main(String[] args)
	{
		if(args.length !=1)
		{
			System.out.println("usage: fib range");
			System.exit(-1);
		}

		System.out.println(calFib(Integer.parseInt(args[0])));

	}
}